### PR TITLE
Use internal-collabPersonId when present

### PR DESCRIPTION
**Pull request by MKodde**
When EB sends the attribute urn:mace:surf.nl:attribute-def:internal-collabPersonId:
* Use this attribute's value as the person's identifier Stepup uses internally.
* Remove this attribute from the set of attributes. It must not be released to the SP in any situation,
* Use the supplied NameID as the NameID to the SP
* Treat the epTID not specially when present. If it's present it will be released to the SP just like another attribute that would be present.

Otherwise, when the attribute is not set:
* Keep current behaviour = Use NameID as person's identifier ,Use epTID as value for NameID for SP, 

Story: https://www.pivotaltracker.com/story/show/181115261

### CODE REVIEW

**Review by Badlapje — APPROVED** [review]
I can't test this functionally, but code-wise this looks great to me.  Here be tests, and code which looks good.  Nice work!

File: src/Surfnet/StepupGateway/GatewayBundle/Tests/Service/ProxyResponseServiceTest.php [hunk 1]
```diff
@@ -155,10 +214,10 @@ public function it_appends_the_proxied_identity_provider_to_existing_authenticat
         );
     }
 
-    public function testCreateProxyResponseRequiresEpti()
+    public function testCreateProxyResponseRequiresEptiIfInternalCollabPersonIdIsNotPresent()
     {
         $this->expectException(RuntimeException::class);
-        $this->expectExceptionMessage('The "urn:mace:dir:attribute-def:eduPersonTargetedID" is not present');
+        $this->expectExceptionMessage('Neither "urn:mace:dir:attribute-def:eduPersonTargetedID" or "urn:mace:surf.nl:attribute-def:internal-collabPersonId" is present');
```
Comment (pmeulen) [inline src/Surfnet/StepupGateway/GatewayBundle/Tests/Service/ProxyResponseServiceTest.php:220]: "Neither" goes with "nor"

File: src/Surfnet/StepupGateway/GatewayBundle/Service/ProxyResponseService.php [hunk 2]
```diff
@@ -110,21 +110,17 @@ public function createProxyResponse(Assertion $assertion, $destination, $loa = n
...
-        // Perform some input validation on the eptiNameId that was received.
-        if (is_null($eptiNameId)) {
-            throw new RuntimeException('The "urn:mace:dir:attribute-def:eduPersonTargetedID" is not present.');
-        } elseif (!array_key_exists(0, $eptiNameId) || !$eptiNameId[0]->value) {
+        $internalCollabPersonId = $translatedAssertion->getAttributeValue('internalCollabPersonId');
+        // Perform some input validation on the eptiNameId and/or internal-collabPersonId that was received.
+        if (is_null($eptiNameId) && is_null($internalCollabPersonId)) {
             throw new RuntimeException(
-                'The "urn:mace:dir:attribute-def:eduPersonTargetedID" attribute does not contain a NameID with a value.'
+                'Neither "urn:mace:dir:attribute-def:eduPersonTargetedID" or ' .
```
Comment (pmeulen) [inline src/Surfnet/StepupGateway/GatewayBundle/Service/ProxyResponseService.php:119]: Typo: "or" => "nor"

**Review by pmeulen — CHANGES_REQUESTED** [review]
Logic looks sound and tests exist.

One small typo to fix.

**Pull request merged**